x-pack/plugins/reporting/server/test_helpers/create_mock_reportingplugin.ts

```diff
@@ -11,88 +11,62 @@ jest.mock('../lib/create_queue');
 jest.mock('../lib/enqueue_job');
 jest.mock('../lib/validate');
 
-import { of } from 'rxjs';
-import { first } from 'rxjs/operators';
-import { coreMock } from 'src/core/server/mocks';
+import * as Rx from 'rxjs';
 import { ReportingConfig, ReportingCore } from '../';
 import {
   chromium,
   HeadlessChromiumDriverFactory,
   initializeBrowserDriverFactory,
 } from '../browsers';
-import { ReportingInternalSetup } from '../core';
-import { ReportingPlugin } from '../plugin';
-import { ReportingSetupDeps, ReportingStartDeps } from '../types';
+import { ReportingInternalSetup, ReportingInternalStart } from '../core';
+import { ReportingStartDeps } from '../types';
 
 (initializeBrowserDriverFactory as jest.Mock<
   Promise<HeadlessChromiumDriverFactory>
 >).mockImplementation(() => Promise.resolve({} as HeadlessChromiumDriverFactory));
 
 (chromium as any).createDriverFactory.mockImplementation(() => ({}));
 
-const createMockSetupDeps = (setupMock?: any): ReportingSetupDeps => {
+const createMockPluginSetup = (setupMock?: any): ReportingInternalSetup => {
   return {
+    elasticsearch: setupMock.elasticsearch || { legacy: { client: {} } },
+    basePath: setupMock.basePath,
+    router: setupMock.router,
     security: setupMock.security,
-    licensing: {
-      license$: of({ isAvailable: true, isActive: true, type: 'basic' }),
-    } as any,
-    usageCollection: {
-      makeUsageCollector: jest.fn(),
-      registerCollector: jest.fn(),
-    } as any,
+    licensing: { license$: Rx.of({ isAvailable: true, isActive: true, type: 'basic' }) } as any,
+  };
+};
+
+const createMockPluginStart = (startMock?: any): ReportingInternalStart => {
+  return {
+    browserDriverFactory: startMock.browserDriverFactory,
+    enqueueJob: startMock.enqueueJob,
+    esqueue: startMock.esqueue,
+    savedObjects: startMock.savedObjects || { getScopedClient: jest.fn() },
+    uiSettings: startMock.uiSettings || { asScopedToClient: () => ({ get: jest.fn() }) },
   };
 };
 
 export const createMockConfigSchema = (overrides?: any) => ({
   index: '.reporting',
-  kibanaServer: {
-    hostname: 'localhost',
-    port: '80',
-  },
-  capture: {
-    browser: {
-      chromium: {
-        disableSandbox: true,
-      },
-    },
-  },
+  kibanaServer: { hostname: 'localhost', port: '80' },
+  capture: { browser: { chromium: { disableSandbox: true } } },
   ...overrides,
 });
 
 export const createMockStartDeps = (startMock?: any): ReportingStartDeps => ({
   data: startMock.data,
 });
 
-const createMockReportingPlugin = async (config: ReportingConfig): Promise<ReportingPlugin> => {
```

Comment:
I'm not sure why originally I decided to create a `ReportingPlugin` object, just for the purpose of extracting the stateful `reportingCore` object out of it. Now `createMockReportingCore` just instantiates a `ReportingCore` on its own.